cranelift/codegen/src/machinst/mod.rs

```diff
@@ -298,6 +299,12 @@ pub trait MachInstEmitState<I: VCodeInst>: Default + Clone + Debug {
     fn on_new_block(&mut self) {}
 }
 
+/// Trait describing what is expected of `EmitInfo` in each backend.
+pub trait MachInstEmitInfo {
+    /// Returns target-independent compilation settings.
+    fn settings(&self) -> &settings::Flags;
```

Comment:
Oh I apparently missed that vcode was already taking a flag from `settings::Flags` so all I had to do was pass in the struct rather than the single flag -- much easier!